### PR TITLE
Load page iframe-api

### PR DESCRIPTION
**Pull request by jonnytest1**
redo of https://github.com/thecodingmachine/workadventure/pull/922 because of accidental commits on the wrong branch

pull request allows loading a new map by an api call

### CODE REVIEW

**Comment by jonnytest1** [comment]
> @Kharhamel  Could you please take a quick look at this fix?

lets see if this speeds things up 😁 (https://github.com/thecodingmachine/workadventure/pull/943#issuecomment-827732799)

**Comment by moufmouf** [comment]
Hey @jonnytest1 ,

Thanks a lot for this PR!

I'm completely :+1: to add this feature in the API.

A few comments:

- We need to find a better method name: the distinction between `loadPage` and `goToPage` is not obvious. We need a better vocabulary. What this is doing is actually taking someone to another room. Maybe we could call the method `goToRoom`, or `exitToRoom`?
- Could you add a test map? In the "maps/tests" folder. There are already a few maps here. You could add a very small map with a small script that triggers the method (and then, reference this test script in `maps/tests/index.html`.

Thanks!

**Comment by jonnytest1** [comment]
🤔 we could rename it "exitScene"/"exitSceneTo" to conform with the attribute people can add to their layer ?
(map comes in a minute)

**Comment by jonnytest1** [comment]
i just added another option to the gotoPage.json

**Comment by jonnytest1** [comment]
oh PS:  /// <reference path="../../front/src/iframe_api.ts" /> allows typeschecking/connection in javascript files

**Comment by jonnytest1** [comment]
resolved conflicts

**Comment by jonnytest1** [comment]
@Kharhamel @moufmouf  just merged develop in so not is good timing to merge this (no conflicts)

**Comment by Kharhamel** [comment]
Hello, sorry we stopped reviewing PR for a while. We just need you to update the documentation with your new feature
The file is here: https://github.com/thecodingmachine/workadventure/blob/develop/docs/maps/api-reference.md

File: front/src/iframe_api.ts [hunk 1]
```diff
@@ -166,6 +168,15 @@ window.WA = {
         }, '*');
     },
 
+    exitSceneTo(url : string) : void{
```
Comment (Kharhamel) [inline front/src/iframe_api.ts:171]: Could you rename this to 'goToRoom'?

Comment (jonnytest1) [reply front/src/iframe_api.ts]: done

Comment (jonnytest1) [reply front/src/iframe_api.ts]: the idea was to hint developers that it does the same as exitSceneUrl

**Comment by jonnytest1** [comment]
> Hello, sorry we stopped reviewing PR for a while. We just need you to update the documentation with your new feature
> The file is here: https://github.com/thecodingmachine/workadventure/blob/develop/docs/maps/api-reference.md

its updated

**Comment by moufmouf** [comment]
Excellent! Thanks a lot!

**Pull request merged**